Chapter 제목 스타일 패치

thekpaul opened the pull request:
졸업논문을 쓰던 중 몇 가지 바꿔본 게 있는데, 혹시 본가에 도움이 될까 싶어 들고 와 봤습니다.

### Chapter 제목 한 줄로 맞추기
![image](https://github.com/Zeta611/snu-ece-bsc-thesis/assets/44695374/947ea324-4aa5-45d6-a44d-37482baa3aa2)


![image](https://github.com/Zeta611/snu-ece-bsc-thesis/assets/44695374/3ee0aafb-7967-4b94-aa82-b1bfe5422b66) **→** ![image](https://github.com/Zeta611/snu-ece-bsc-thesis/assets/44695374/2e45048c-f932-4d09-9cce-01c0fd34e9bf)
크기 정도만 바꾸면 더 비슷해질 것 같긴 한데 이대로 둬도 문제는 없지 않을까 싶어요

### Chapter 제목 위치 (["3센티 드랍"][1])
![image](https://github.com/Zeta611/snu-ece-bsc-thesis/assets/44695374/d4b450d0-4f14-4cc6-8530-f99aa2c0d9a2)
페이지 상단에서 3센티 드랍을 시키는 부분입니다.

![image](https://github.com/Zeta611/snu-ece-bsc-thesis/assets/44695374/8a302bfc-7480-4194-a034-0a384076b247) **→** ![image](https://github.com/Zeta611/snu-ece-bsc-thesis/assets/44695374/1cdc9cba-beec-48fc-b278-6bd13d8247b3)
`titlesec` 패키지를 추가로 받아서 드랍시키고, 언어에 따라서 Chapter 제목도 바뀌는만큼 이 부분을 `ko` 옵션 받는 위치에 넣어서 따로 해봤습니다.

더 바꿀 부분 있으면 추천받습니다.
이번엔 꼭 졸업하고 싶습니다.
흑흑

[1]: https://namu.wiki/w/3%EC%84%BC%ED%8B%B0%20%EB%93%9C%EB%9E%8D

thekpaul committed:
FEAT: Add Default Styles for Abstract Titles

Centered titles and a 3-centimeter horizontal skip for Korean
is specified in the ECE standard template.
To ensure that centering and horizontal skip do not affect the ToC,
this commit introduces constants `\l_snu_ece_bsc_thesis_abstractstyle_*`
for each supported language.
These "style" constants are used to typeset chapter titles conformant to
the provided standards, while the original "name" constants are passed
to the ToC line.

To further support abstract environments without an explicit language
option, an extra string comparison condition is implemented.
This condition checks if the `\abstractname` has been modified -
if not, the predefined "name" constants are passed to the ToC by default
to mimic behaviour for environments with an explicit language option
provided.
If the `\abstractname` has been modified, the modified `\abstractname`
is passed as-is to the ToC.

Signed-off-by: Paul Kim <44695374+thekpaul@users.noreply.github.com>

thekpaul committed:
FEAT: Remove Vertical Skips and Center Chapters

In further accordance with the ECE standards, chapter titles must be
offset by 3 centimeters from the top of the page,
with its contents centered and typeset in a single line.
The current base class `report` does not support such changes without
modifying the underlying macros, effectively "hard-coding" some aspects
that require changes.

To circumvent this problem, a new package `titlesec` is introduced as a
prerequisite for this class;
macros `\titlespacing` and `\titleformat` are used to modify vertical
spacing and formatting, respectively.

This removes the necessity for abstract titles to be centered
explicitly, as their `\chapter*{...}`s are also centered by default
with the new `\titleformat` in effect.

Note that the `\titlespacing` macro has no effect on `\chapter`s unless
the `\titleformat` macro is also used.

Signed-off-by: Paul Kim <44695374+thekpaul@users.noreply.github.com>

Zeta611 commented on snu-ece-bsc-thesis.cls:
```diff
@@ -261,8 +278,28 @@
       { % No language specified
-        \chapter*{\abstractname}
-        \addcontentsline{toc}{chapter}{\abstractname}
+        \ifthenelse {%
```
일관성을 위해 `\tl_if_eq:nnTF` 방식으로 변경해주실 수 있을까요? (`texdoc interface3`, 15.3 Token list conditionals)
그리고 expl3 맥락 안에서는 %가 없어도 됩니다. (모든 공백 무시)
```tex
\tl_if_eq:NNTF \abstractname \l_snu_ece_bsc_thesis_abstractstyle_ko
  { % Abstract name is ...
    ...
  }
  {
    \tl_if_eq:NNTF \abstractname \l_snu_ece_bsc_thesis_abstractstyle_en
      { ...
      }
      { ...
      }
  }
```

thekpaul replied:
Commit a95a1a0:

```diff
@@ -278,18 +278,14 @@
           }
       }
       { % No language specified
-        \ifthenelse {%
-          \equal{\abstractname}{\l_snu_ece_bsc_thesis_abstractstyle_ko}%
-        }
+        \tl_if_eq:NNTF \abstractname \l_snu_ece_bsc_thesis_abstractstyle_ko
           { % Abstract name is unchanged in Korean environment
             \chapter*{\abstractname}
             \addcontentsline{toc}{chapter}
               {\text_purify:V\l_snu_ece_bsc_thesis_abstractname_ko}
           }
           {
-            \ifthenelse {%
-              \equal{\abstractname}{\l_snu_ece_bsc_thesis_abstractstyle_en}%
-            }
+            \tl_if_eq:NNTF \abstractname \l_snu_ece_bsc_thesis_abstractstyle_en
               { % Abstract name is unchanged in English environment
                 \chapter*{\abstractname}
                 \addcontentsline{toc}{chapter}
```

Zeta611 commented on snu-ece-bsc-thesis.cls:
```diff
@@ -82,10 +92,17 @@
+    \RequirePackage{titlesec}
+
+    \titlespacing*{\chapter}{0pt}{-10mm}{10mm}
+    \titleformat{\chapter}[block]{\centering \huge \bfseries}%
```
No trailing % required in an expl3 context.

thekpaul replied:
Commit a95a1a0:

```diff
@@ -76,7 +76,7 @@
     \RequirePackage{titlesec}

     \titlespacing*{\chapter}{0pt}{-10mm}{10mm}
-    \titleformat{\chapter}[block]{\centering \huge \bfseries}%
+    \titleformat{\chapter}[block]{\centering \huge \bfseries}
                 {Chapter\ \thechapter}{22pt}{}
   }
   { % Korean main
@@ -101,7 +101,7 @@
     \RequirePackage{titlesec}

     \titlespacing*{\chapter}{0pt}{-10mm}{10mm}
-    \titleformat{\chapter}[block]{\centering \huge \bfseries}%
+    \titleformat{\chapter}[block]{\centering \huge \bfseries}
                 {제\ \thechapter\ 장}{20pt}{}
   }
```

Zeta611 reviewed (changes requested):
순탄한 졸업을 기원합니다 😄

thekpaul committed:
REFACTOR: Conform to Consistent `expl3` Syntax Use

Applies the `\tl_if_eq:NNTF` macro in place of `\ifthenelse` for
consistency in using `expl3` syntax.
Also removes trailing `%`, as they are unnecessary in `expl3` context.

Signed-off-by: Paul Kim <44695374+thekpaul@users.noreply.github.com>

Zeta611 reviewed (approved):
감사합니다!!
